interpreter/valid/valid.ml

```diff
@@ -496,64 +495,35 @@ let rec check_instr (c : context) (e : instr) (s : infer_result_type) : infer_in
     [RefT (NoNull, DefHT y)] -->
     [RefT (NoNull, DefHT (Stat x.it))], []
 
-  | ContBind x ->
-    (match peek_ref 0 s e.at with
-    | nul, DefHT (Stat y) ->
-      let ContT z = cont_type c (y @@ e.at) in
-      let FuncT (ts1, ts2) = func_type c (as_stat_var z @@ e.at) in
-      let ContT z' = cont_type c x in
-      let FuncT (ts1', _) as ft' = func_type c (as_stat_var z' @@ x.at) in
-      require (List.length ts1 >= List.length ts1') x.at
-        "type mismatch in continuation arguments";
-      let ts11, ts12 = Lib.List.split (List.length ts1 - List.length ts1') ts1 in
-      require (match_func_type c.types (FuncT (ts12, ts2)) ft') e.at
```

Comment:
It appears you removed this, but I believe a check along these lines is still necessary, otherwise x and y could be completely unrelated.